source/SpinalHDL/Structuring/parametrization.rst

```diff
@@ -46,6 +84,21 @@ For optional signal :
     val mySignal = flag generate (Bool())  //Equivalent to "if (flag) in Bool() else null"
   }
 
+.. note::
+
```

Comment:
- `generate` is a method/function, not a "keyword like directive" - `flag.generate(Bool())` is exactly the same
- "follows for a value" seemed like a typo to me (reading it a few times I get what you want to say) + this is written quite complicated. How about something along the lines of "generate will evaluate the given expression depending on the predicate, returning `null` if the predicate is false"?

Reply:
Technically everything is a method/function, no code exists in the JVM that doesn't follow that paradigm of a bytecode entry point.
So the question then is, what would you want to initially teach someone who does not know anything about the subject ?
Can this be done in a way that they grasp the new concept enough to understand it and move to the next thing.
I find the term `keyword like directive` to encapsulate the nature of this one, it looks and acts like the keyword, it is controlling behaviour like a directive, the syntax is expected to be used like a keyword.   The term stays in the domain of describing how to use it, and not what it does.

Agree on the "follows for a value" is not a good explanation, I am trying to emphasis the expression must return a value, not `Unit` / void.

With your wording, it sounds funny to start the explanation about the thing that happens 2nd (evaluation of expression when predicate true).  So starting the explanation with the evaluation of the predicate should be first, which also follows how it is evaluated if there are additional side-effects (other than the value returned) during the optionally evaluated expression when predicate true.  Changing the order/rewording would solve that.

Maybe it can be concisely explained in terms of a `Ternary` operator concepts ? 

The first sentence (or so) is trying to formally define what is does in terms and rules that are easy to work through and difficult to misunderstand.  I don't know if that adds to the perceived complexity of the explanation.

Reply:
You might find "keyword like directive" to be a good term, but this is not used anywhere else in either Scala or SpinalHDL docs so googling for it you find no helpful search results. Inventing new names will not make this any easier - and might obfuscate what is actually happening.

Back to the wording, let's turn it around then: "If the predicate is true, generate will evaluate the given expression and return the result, otherwise is returns null"?

Reply:
I like the term it conveys so much to me.

Give this version a go.

I think everything in Scala returns a value, it is the Monadic way, even Unit and null is a value.
But the purpose of referencing it is to make the user think about the value being returned, but maybe that is a given from the code example and expected Scala understanding of the reader.